index.js

```diff
@@ -1,2 +1,601 @@
+'use strict';
+
+var base32 = require('base32.js');
+var crypto = require('crypto');
+var url = require('url');
+var util = require('util');
+
+/**
+ * Digest the one-time passcode options.
+ *
+ * @param {Object} options
+ * @param {String} options.secret Shared secret key
+ * @param {Integer} options.counter Counter value
+ * @param {String} [options.encoding="ascii"] Key encoding (ascii, hex,
+ *   base32, base64).
+ * @param {String} [options.algorithm="sha1"] Hash algorithm (sha1, sha256,
+ *   sha512).
+ * @param {String} [options.key] (DEPRECATED. Use `secret` instead.)
+ *   Shared secret key
+ * @return {Buffer} The one-time passcode as a buffer.
+ */
+
+exports.digest = function digest (options) {
+  var i;
+
+  // unpack options
+  var key = options.secret;
+  var counter = options.counter;
+  var encoding = options.encoding || 'ascii';
+  var algorithm = (options.algorithm || 'sha1').toLowerCase();
+
+  // Backwards compatibility - deprecated
+  if (options.key) {
+    console.log('Speakeasy - Deprecation Notice - Specifying the secret using `key` is no longer supported. Use `secret` instead.');
+    key = options.key;
+  }
+
+  // convert key to buffer
+  if (!Buffer.isBuffer(key)) {
+    key = encoding === 'base32' ? base32.decode(key)
+      : new Buffer(key, encoding);
+  }
+
+  // create an buffer from the counter
+  var buf = new Buffer(8);
+  var tmp = counter;
+  for (i = 0; i < 8; i++) {
+    // mask 0xff over number to get last 8
+    buf[7 - i] = tmp & 0xff;
+
+    // shift 8 and get ready to loop over the next batch of 8
+    tmp = tmp >> 8;
+  }
+
+  // init hmac with the key
+  var hmac = crypto.createHmac(algorithm, key);
+
+  // update hmac with the counter
+  hmac.update(buf);
+
+  // return the digest
+  return hmac.digest();
+};
+
+/**
+ * Generate a counter-based one-time token.
+ *
+ * @param {Object} options
+ * @param {String} options.secret Shared secret key
+ * @param {Integer} options.counter Counter value
+ * @param {Buffer} [options.digest] Digest, automatically generated by default
+ * @param {Integer} [options.digits=6] The number of digits for the one-time
+ *   passcode.
+ * @param {String} [options.encoding="ascii"] Key encoding (ascii, hex,
+ *   base32, base64).
+ * @param {String} [options.algorithm="sha1"] Hash algorithm (sha1, sha256,
+ *   sha512).
+ * @param {String} [options.key] (DEPRECATED. Use `secret` instead.)
+ *   Shared secret key
+ * @param {Integer} [options.length=6] (DEPRECATED. Use `digits` instead.) The
+ *   number of digits for the one-time passcode.
+ * @return {String} The one-time passcode.
+ */
+
+exports.hotp = function hotpGenerate (options) {
+  // unpack digits
+  // backward compatibility: `length` is also accepted here, but deprecated
+  var digits = (options.digits != null ? options.digits : options.length) || 6;
+  if (options.length) console.log('Speakeasy - Deprecation Notice - Specifying token digits using `length` is no longer supported. Use `digits` instead.');
```

Comment:
I did `if (options.length)` as a shortcut since 0 would also be an error, but that's true – 0 doesn't really fall under this. I'll have to do something like `if (options.length !== null || typeof options.length !== 'undefined')` since Semistandard enforces `===` and `!==`.